components/tools/annotation.js

```diff
@@ -3,95 +3,53 @@
-
-        var root = null,
-            line = null,
-            callout = null;
-
         var annotation = function(selection) {
-
-            root = selection.append('g')
-                .attr('id', 'annotation_' + index)
-                .attr('class', 'annotation');
-
-            line = root.append('line')
-                .attr('class', 'marker')
-                .attr('x1', xScale.range()[0])
-                .attr('y1', yScale(yValue))
-                .attr('x2', xScale.range()[1])
-                .attr('y2', yScale(yValue));
-
-
-            callout = root.append('text')
-                .attr('class', 'callout')
-                .attr('x', xScale.range()[1] - padding)
-                .attr('y', yScale(yValue) - padding)
-                .attr('style', 'text-anchor: end;')
-                .text(yLabel + ': ' + formatCallout(yValue));
-        };
-
-        annotation.index = function(value) {
-            if (!arguments.length) {
-                return index;
-            }
-            index = value;
-            return annotation;
-        };
-
-        annotation.xScale = function(value) {
-            if (!arguments.length) {
-                return xScale;
-            }
-            xScale = value;
-            return annotation;
-        };
-
-        annotation.yScale = function(value) {
-            if (!arguments.length) {
-                return yScale;
-            }
-            yScale = value;
-            return annotation;
-        };
-
-        annotation.yValue = function(value) {
-            if (!arguments.length) {
-                return yValue;
-            }
-            yValue = value;
-            return annotation;
-        };
-
-        annotation.yLabel = function(value) {
-            if (!arguments.length) {
-                return yLabel;
-            }
-            yLabel = value;
-            return annotation;
-        };
-
-        annotation.padding = function(value) {
-            if (!arguments.length) {
-                return padding;
-            }
-            padding = value;
-            return annotation;
-        };
-
-        annotation.formatCallout = function(value) {
-            if (!arguments.length) {
-                return formatCallout;
-            }
-            formatCallout = value;
-            return annotation;
-        };
+            selection.each(function(data) {
+                var x = annotation.xScale.value,
+                    y = annotation.yScale.value,
+                    yValue = annotation.yValue.value;
+
+                var container = d3.select(this);
+
+                // Create a group for each annotation
+                var g = fc.utilities.simpleDataJoin(container, 'annotation', data, yValue);
+
+                // Added the required elements - each annotation consists of a line and text label
+                var enter = g.enter();
+                enter.append('line')
+                    .attr('class', 'marker');
```

Comment:
Sure, just elements, without classes, would work here.

My initial though was, if the elements had class names, it might be easier for the user if they were making use of decorate. For example, if they wanted to append an additional line, they may not want the default annotation styling applied (I know there are easy ways around this, but it could be annoying for the user). I suppose this is an edge case and wouldn't affect the vast majority of use cases.

I agree that using just element is more appropriate here - I'll make the change.